security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -521,6 +523,21 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 		certChain = append(certChain, []byte(c)...)
 	}
 
+	// Cert exipre time by default is createTime + sc.configOptions.SecretTTL.
+	expireTime := t.Add(sc.configOptions.SecretTTL)
+	block, _ := pem.Decode(certChain)
+	if block != nil {
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err == nil {
+			expireTime = cert.NotAfter
+			log.Debugf("Cert expire time is %v", expireTime)
+		} else {
+			log.Debugf("Failed to parse certificate %v, use default cert TTL", err.Error())
```

Comment:
throw Errorf instead?

Reply:
current logic is CA respect secret TTL passed from nodeagent and uses it as cert TTL, this PR is from ask in https://github.com/istio/istio/issues/12674, the updated logic is
1. nodeagent parse cert to get TTL(to handle case that customer CA don't respect TTL passed to it)
2. fallback to default configured TTL if cert parse fail(for some reason)
so I think using debug log is fine.